lib/flay/commands/release.rb

```diff
@@ -0,0 +1,84 @@
+class Flay::Commands::Release < Thor::Group
+  include Flay::Helpers
+  include Thor::Actions
+
+  ERROR_METADATA = "metadata.rb not found".freeze
+  ERROR_GIT      = "There are files that need to be committed before releasing".freeze
+  ERROR_VERSION  = "Missing or invalid version in metadata.rb".freeze
+
+  VERSION_MATCH = /^\s*version\s*\(?'?"?([^"']+)"?'?\)?\s*$/.freeze
+
+  def self.register_with(thor, as: "MISSING NAME")
+    thor.register(self, as, as, "creates a release for this cookbook")
+  end
+
+  def self.exit_on_failure?
+    true
+  end
+
+  desc "creates a new release and uploads it"
+
+  def fail_fast
+    exit 1 unless git_clean? && git_committed?
+    exit 1 unless metadata_exists?
+    exit 1 if version.nil?
+  end
+
+  def berks_install
+    say "Installing berks dependencies...", :green
+    shell_exec("chef exec berks install")
+  end
+
+  def create_tag
+    return if tag_exists?
+    say "Creating tag for v#{version}...", :green
+    shell_exec("git tag -a -m \"Version #{version}\" v#{version}")
+  end
+
+  def push_commits_and_tags
+    say "Pushing commits and tags...", :green
+    fail Thor::Error, "Couldn't push commits" unless shell_exec("git push", show_output: false).last == 0
+    fail Thor::Error, "Couldn't push tags" unless shell_exec("git push --tags", show_output: false).last == 0
```

Comment:
Ugh...thor returns with 0 even if an exception is raised. Should `exit 1` here instead of raising